src/renault_api/cli/charge/schedule.py

```diff
@@ -123,6 +123,63 @@ async def set(
     click.echo(write_response.raw_data)
 
 
+async def _set_schedule_activated(
+    ctx_data: Dict[str, Any],
+    websession: aiohttp.ClientSession,
+    id: int,
+    activated_state: bool,
+) -> None:
+    """Set the given schedules activated-flag to given state."""
+    vehicle = await renault_vehicle.get_vehicle(
+        websession=websession, ctx_data=ctx_data
+    )
+    response = await vehicle.get_charging_settings()
+
+    if not response.schedules:  # pragma: no cover
+        click.echo("No schedules found.")
+        return
+
+    id_found = False
+    for schedule in response.schedules:
+        if id == schedule.id:  # pragma: no cover
+            schedule.activated = activated_state
+            id_found = True
+            break
+    if not id_found:
+        raise IndexError(f"Schedule id {id} not found.")  # pragma: no cover
```

Comment:
Can this be replaced by a `_get_schedule` method, which returns a `Tuple(vehicle, schedule)` or raises exception on failure.

Then the `activate`/`deactivate`/`set` methods can use `_get_schedule` instead of `get_vehicle`